src/MongoDB.Driver.Core/Core/Operations/ChangeStreamCursor.cs

```diff
@@ -25,20 +25,41 @@
 
 namespace MongoDB.Driver.Core.Operations
 {
+    /// <summary>
+    /// Represents an asynchronous cursor for change stream.
+    /// </summary>
+    /// <typeparam name="TDocument"></typeparam>
+    public interface IChangeStreamCursor<out TDocument> : IAsyncCursor<TDocument>
+    {
+        /// <summary>
+        /// Gets the resume token.
+        /// </summary>
+        /// <value>
+        /// The resume token.
+        /// </value>
+        BsonDocument GetResumeToken();
+    }
+
     /// <summary>
     /// A change stream cursor.
     /// </summary>
     /// <typeparam name="TDocument">The type of the output documents.</typeparam>
     /// <seealso cref="MongoDB.Driver.IAsyncCursor{TOutput}" />
-    internal sealed class ChangeStreamCursor<TDocument> : IAsyncCursor<TDocument>
+    internal sealed class ChangeStreamCursor<TDocument> : IChangeStreamCursor<TDocument>
     {
         // private fields
         private readonly IReadBinding _binding;
-        private readonly IChangeStreamOperation<TDocument> _changeStreamOperation;
+        private IChangeStreamOperation<TDocument> _changeStreamOperation;
         private IEnumerable<TDocument> _current;
         private IAsyncCursor<RawBsonDocument> _cursor;
         private bool _disposed;
-        private IBsonSerializer<TDocument> _documentSerializer;
+        private BsonDocument _documentResumeToken;
+        private readonly IBsonSerializer<TDocument> _documentSerializer;
+        private readonly BsonTimestamp _initialOperationTime;
+        private BsonDocument _postBatchResumeToken;
+        private readonly BsonDocument _resumeAfter;
+        private readonly BsonDocument _startAfter;
+        private readonly BsonTimestamp _startAtOperationTime;
```

Comment:
Rename these three variables to

    _initialResumeAfter
    _initialStartAfter
    _initialStartAtOperationTime

to be very clear that these are the initial values supplied by the application and not necessarily the values that will be used when resuming.

Reply:
Done